Parent/Parent/ParentAppDelegate.swift

```diff
@@ -269,11 +277,24 @@ extension ParentAppDelegate: AnalyticsHandler {
 //        Analytics.logEvent(name, parameters: parameters)
     }
 
-    private func initializeHeap() {
-        guard !ProcessInfo.isUITest, let heapID = Secret.heapID.string else { return }
+    private func initializeTracking() {
+        guard
+            let environmentFeatureFlags,
+            !ProcessInfo.isUITest,
+            let heapID = Secret.heapID.string
+        else {
+            return
+        }
+
+        let isSendUsageMetricsEnabled = environmentFeatureFlags.isFeatureEnabled(.send_usage_metrics)
         let options = HeapOptions()
-        options.disableAdvertiserIdCapture = true
+        options.disableTracking = !isSendUsageMetricsEnabled
         Heap.initialize(heapID, with: options)
+        Heap.setTrackingEnabled(isSendUsageMetricsEnabled)
+    }
+
+    private func disableTracking() {
```

Comment:
Reporting doesn't get disabled when a user is changed. `environmentFeatureFlags?.refresh(force: true)` will return the last value in CoreData in case the API request fails. I suggest to turn reporting on only if we receive a go from the API.

Reply:
Changed.